src/gtnh/technic.py

```diff
@@ -0,0 +1,177 @@
+        os.makedirs(mod_dir / "mods")
+        copy(mod_file_path, mod_dir / "mods")
+
+        # zipping with technic format
+        with ZipFile(mod_dir / f"{mod_name}-{mod.version}.zip", "w") as mod_archive:
+            for x in (mod_dir / "mods").iterdir():
+                if x.is_file():
+                    mod_archive.write(x, x.relative_to(mod_dir))
+
+        # removing temp dir structure
+        rmtree(mod_dir / "mods")
+
+    # handling of the modpack repo
+    if callback is not None:
+        callback(delta_progress, "generating technic assets for additional modpack files")
+    # path for the already made client dev pack archive
+    modpack_folder = ensure_cache_dir() / "client_archive"
+
+    # zipping of the archive
+    os.makedirs(destination / "gtnhmodpack", exist_ok=True)
+    log.info("Processing gtnh modpack assets.")
+    with ZipFile(destination / "gtnhmodpack" / f"gtnhmodpack-{modpack_version}.zip", "w") as mod_archive:
+
+        def crawl_zip(path: Path) -> None:
+            for x in path.iterdir():
+                if x.is_dir():
+                    crawl_zip(x)
+                elif x.is_file() and not x.name.endswith(".jar"):  # excluding all the jar files as we use the client dev
+                    # pack folder that is already filled with mod jars
+                    mod_archive.write(x, x.relative_to(modpack_folder))
+                    log.info(f"zipped {x.relative_to(modpack_folder)}")
+
+        crawl_zip(modpack_folder)
+
+    # handling forge asset
+    if callback is not None:
+        callback(delta_progress, "generating technic assets for forge")
+
+    forge_path = destination / "modpack" / "modpack-1.7.10-10.13.4.1614.zip"
+    os.makedirs(forge_path.parent)
+    log.info("downloading forge assets")
+    download_file("http://downloads.gtnewhorizons.com/DreamAssemblerXXL/Technic/modpack-1.7.10-10.13.4.1614.zip", forge_path)
+
+    log.info(f"successfully finished the assembling of technic assets. Folder availiable for upload at {destination}")
+
+
+def get_mod_name(mod_name: str) -> str:
+    """
+    takes the mod name and process it so it has only lower alphanumerical chars
+    :return: the normalised mod name
+    """
+
+    return re.sub("[^a-zA-Z0-9]", "", mod_name).lower()
+
+
+def download_file(url: str, path: Path, headers: Optional[Dict[str, str]] = None) -> None:
+    """
+    Downloads a file from the url and save it to path
+
+    :param url: specified url
+    :param path: specified path
+    :param headers: optional headers when dowloading the file
+    :return: None
+    """
+    if headers is None:
+        headers = {"Accept": "application/octet-stream"}
+
+    with requests.get(url, stream=True, headers=headers) as r:
+        r.raise_for_status()
+        with open(path, "wb") as f:
+            for chunk in r.iter_content(chunk_size=8192):
+                f.write(chunk)
+
+
+if __name__ == "__main__":
+    process_files("2.1.2.4")
```

Comment:
nit: should use click() or something so it can be called from the command line (CI) with an argument